Client/mods/deathmatch/logic/CClientGame.cpp

```diff
@@ -5428,16 +5428,8 @@ void CClientGame::ResetMapInfo()
 
     m_bHudAreaNameDisabled = false;
 
-    // Gravity
-    g_pMultiplayer->SetLocalPlayerGravity(DEFAULT_GRAVITY);
-    g_pMultiplayer->SetGlobalGravity(DEFAULT_GRAVITY);
-    g_pGame->SetGravity(DEFAULT_GRAVITY);
-
-    // Gamespeed
-    SetGameSpeed(DEFAULT_GAME_SPEED);
-
-    // Game minute duration
-    SetMinuteDuration(DEFAULT_MINUTE_DURATION);
+    // Reset world special properties, world properties, weather properties etc
+    ResetWorldProperties(true, true, true, true, true);
```

Comment:
It would be better to pass a struct instead of unnamed 5 booleans. For example:

```
struct ResetWorldPropsDesc
{
    bool resetSpecialProperties;
    bool resetWorldProperties;
    bool resetWeatherProperties;
    bool resetLODs;
    bool resetSounds
}
```

And since C++20(I hope we will have it in the future) you will be able to:

```
ResetWorldProperties({
    .resetSpecialProperties = true,
    .resetWorldProperties = false,
    .resetWeatherProperties = true,
    ...
})
```

But for now you can just

`ResetWorldProperties({true, true, true, ...})`

Reply:
> More than 2 or 3 arguments should be represented as a struct. It's not a part of guidlines yet but it should be I think.

That would complicate a lot of functions. for more than 5 arguments I think its fine but it still casts a huge shadow of doubt to it

Reply:
At some point, when number of arguments continues to increase, the doubts of using struct will become less than doubts of using arguments in a row. That's the point. This is the standart practice in most projects.